lib/distributed_cache.rb

```diff
@@ -3,6 +3,36 @@
 require 'message_bus/distributed_cache'
 
 class DistributedCache < MessageBus::DistributedCache
+  class << self
+    module Test
+      def caches
+        @caches ||= {}
+      end
+
+      def get(key, **opts)
+        caches[key] ||= new(key, **opts)
+      end
+
+      def clear_caches!
+        caches.each_value do |cache|
```

Comment:
hmmm `@caches = {}` ?

Reply:
In this version, the classes still have instance variables that refer to caches.

Reply:
And there's one cache that's [not a class instance variable too](https://github.com/discourse/discourse/blob/master/lib/active_record/connection_adapters/postgresql_fallback_adapter.rb#L17), which is why I prefer it this way if possible.